### PR TITLE
add network model

### PR DESCRIPTION
**Pull request by tgreen24**

### CODE REVIEW

File: Movies/Models/MovieModel.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
+//
+//  MovieModel.swift
+//  Movies
+//
+//  Created by Tyler Green on 5/12/23.
+//
+
+import Foundation
+
+struct MovieModel: Codable {
```
Comment (daveddm) [inline Movies/Models/MovieModel.swift:14]: So the json response for the `search` endpoint has a key `results` which maps to an array of json objects (the movies). For codable to work you will need another object that represents the outer object. So something like 
```
struct SearchResults: Codable {
    let results: [MovieModel]
}
```

When you make the network call the `SearchResults` model is what you decode from the response. Then you can just return the array of movie models to the UI layer.

**Review by daveddm — CHANGES_REQUESTED** [review]
Had one comment

File: Movies/Models/MovieModel.swift [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import Foundation
+
+struct SearchResults: Codable {
+    let results: [MovieModel]
+}
+
+struct MovieModel: Codable {
+    let id: Int
+    var title: String
```
Comment (daveddm) [inline Movies/Models/MovieModel.swift]: Is there a scenario you can think of where you would need to change the title of a movie?

File: Movies/Models/MovieModel.swift [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    
+    init(id: Int, title: String) {
```
Comment (daveddm) [inline Movies/Models/MovieModel.swift]: Swift will provide an automatic init for structs so you dont need to implement this unless you are accessing this struct in a separate framework

**Review by daveddm — CHANGES_REQUESTED** [review]
Few more comments

File: Movies/Models/MovieModel.swift (hunk 2 above)
Comment (tgreen24) [reply Movies/Models/MovieModel.swift]: I guess not, are you saying it should be a `let` then?

Comment (daveddm) [reply Movies/Models/MovieModel.swift]: Yep if the property is not something that can change you will want to use `let`

**Review by daveddm — APPROVED** [review]

**Pull request merged**